### ISSUE TITLE
Add instrumentation and entity relationship support for AWS_KINESIS_DELIVERY_STREAM

### ISSUE THREAD

**Issue by kford-newrelic** [newrelic/newrelic-java-agent#2070]
### Description
Add core instrumentation support for the AWS_KINESIS_DELIVERY_STREAM service to the Java Agent, along with necessary linking data for entity relationships.

### Acceptance Criteria
When APM monitored applications are using the AWS_KINESIS_DELIVERY_STREAM service, entity relationships are created.

### Design Consideration/Limitations
Assume the data to be emitted for linking will need to rely on ARN-like matching, requiring the AWS account ID to be available in some manner.

### Dependencies
Will need to leverage work under #2003 to get the needed account ID.

### Additional context
N/A

**Comment by obenkenobi** [comment]
The functionality on the agent side is essentially "code complete" and merged into the main branch of the agent but not done.
Waiting on end to end validation to be completed till the ticket is closed.

**Comment by obenkenobi** [comment]
End to end validation is complete, some more discussions if other agent teams would want to replicate the Java Agent's approach needs to be done until the ticket is closed.
